test/elixir/test/basics_test.exs

```diff
@@ -58,6 +58,62 @@ defmodule BasicsTest do
     assert context[:db_name] in Couch.get("/_all_dbs").body, "Db name in _all_dbs"
   end
 
+  test "Database name with '+' should encode to '+'", _context do
```

Comment:
The tests present in the module do not show that encoding of path part and query part is independent. 
It would be better to add an additional check into each test case. It seems you can use `GET /{db}/_changes?filter=_doc_ids&doc_ids=["hello+world"]'`. Where used `doc_ids` would have `+` in there.